### DISCUSSION TITLE
Is it possible to automatically replace all instances of a sequence of characters in an entered command with some other string?

### DISCUSSION THREAD

**Discussion by laser** [scandum/tintin#95] [Script help]
Howdy!

When I enter a command, I would like to replace all instances of some substring to some other string in commands that I'd entered.

To that end, I created a macro like so:

```
#MACRO {xyz} {somereallylongstringhere}
```

Unfortunately, that doesn't appear to be how macros work (`somereallylongstringhere` is sent to the server immediately after I enter the sequence `xyz`).

Some examples of what I would like to enter into my client (and what I would like the server to receive):

| Command I Enter                       | Server Receives                                                                 |
|---------------------------------------|---------------------------------------------------------------------------------|
| give shield xyz                       | give shield somereallylongstringhere                                            |
| tell xyz your character's name is xyz | tell somereallylongstringhere your character's name is somereallylongstringhere |
| look xyz                                  | look somereallylongstringhere                                                   |

I've been thus far unsuccessful at locating a portion of the manual where this is explained. It's likely that I'm missing something - if so, please do feel free to tell me to RTFM better.

Take care, and thanks for the great (free) software!

**Comment by scandum** [comment]
The typical solution would be:
```
#var {xyz} {somereallylongstringhere}
give shield $xyz
```
If you really want to go the macro route you could try:
```
#MACRO {xyz} {#cursor set somereallylongstringhere}
```

**Comment by laser** [comment]
@scandum - Excellent! That works just fine. Thanks for the help.
